repos/fdbt-site/src/pages/capConfirmation.tsx

```diff
@@ -0,0 +1,255 @@
+            <>
+                <h1 className="govuk-heading-l">Check your answers before submitting your fares information</h1>
+                <ConfirmationTable
+                    header="Cap Information"
+                    confirmationElements={buildCapConfirmationElements(
+                        typeOfCap,
+                        productGroupName,
+                        caps,
+                        capValidity,
+                        capStartInfoContent,
+                        services,
+                        tapsPricingContents,
+                        capDistancePricingContents,
+                        distanceBands,
+                    )}
+                />
+                <input type="submit" value="Continue" id="continue-button" className="govuk-button" />
+            </>
+        </CsrfForm>
+    </TwoThirdsLayout>
+);
+
+export const getServerSideProps = async (ctx: NextPageContextWithSession): Promise<{ props: CapConfirmationProps }> => {
+    const csrfToken = getCsrfToken(ctx);
+    const noc = getAndValidateNoc(ctx);
+
+    const typeOfCapAttribute = getSessionAttribute(ctx.req, TYPE_OF_CAP_ATTRIBUTE);
+
+    if (!typeOfCapAttribute || !('typeOfCap' in typeOfCapAttribute)) {
+        throw new Error('Could not extract the correct attributes for the user journey.');
+    }
+
+    const capAttribute = getSessionAttribute(ctx.req, CAPS_ATTRIBUTE);
+    const capStartAttribute = getSessionAttribute(ctx.req, CAP_START_ATTRIBUTE);
+
+    const productGroupIdAttribute = getSessionAttribute(ctx.req, CAPPED_PRODUCT_GROUP_ID_ATTRIBUTE);
+
+    const productGroupName =
+        productGroupIdAttribute && typeof productGroupIdAttribute === 'string'
+            ? (await getProductGroupByNocAndId(noc, Number.parseInt(productGroupIdAttribute)))?.name
+            : '';
+
+    const caps = capAttribute ? capAttribute.caps : [];
+
+    const capValidityAttribute = getSessionAttribute(ctx.req, CAP_EXPIRY_ATTRIBUTE);
+    const capValidity =
+        capValidityAttribute && isCapExpiry(capValidityAttribute)
+            ? `${capValidityAttribute.productValidity}${
+                  capValidityAttribute.productEndTime ? ` - ${capValidityAttribute.productEndTime}` : ''
+              }`
+            : '';
+
+    const serviceListAttribute = getSessionAttribute(ctx.req, SERVICE_LIST_ATTRIBUTE);
+
+    const multiTapsPricingAttribute = getSessionAttribute(ctx.req, MULTI_TAPS_PRICING_ATTRIBUTE);
+
+    const capDistancePricingAttribute = getSessionAttribute(ctx.req, CAP_PRICING_PER_DISTANCE_ATTRIBUTE);
+
+    const tapsPricingContents: string[] = [];
+    if (multiTapsPricingAttribute && !isWithErrors(multiTapsPricingAttribute)) {
+        Object.entries(multiTapsPricingAttribute.tapDetails).forEach((tap) => {
+            tapsPricingContents.push(`Tap number: ${Number(tap[0]) + 1}, Price: ${tap[1]}`);
+        });
+    }
+
+    let capDistancePricingContents: string[] = [];
+    const distanceBands: string[] = [];
```

Comment:
If I select 'rolling days' in /defineCapStart, it is not shown in the confirmation page.

Reply:
Also, if the user creates a 'capped by taps' product, and clicks the back button whilst on the caps confirmation page, the site errors.